pkg/object/strings.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package object
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/yaml"
+)
+
+var codec = scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
```

Comment:
This same codec line already exists in the testutil and 2 other tests. But since YamlStringer is really only for logs, I can change it to just use `sigs.k8s.io/yaml` like we use elsewhere.

As for `sigs.k8s.io/kustomize/kyaml`, does it provide any benefit over `sigs.k8s.io/yaml` or `k8s.io/apimachinery/pkg/util/yaml` for Marshal? Or is it just needed for Parse to RNode?

Reply:
Answering my own question: 
`sigs.k8s.io/kustomize/kyaml`.Marshal is just an alias to `gopkg.in/yaml.v3`.Marshal

Reply:
Does using the `JsonFallbackEncoder` necessarily lead to a dependency on the concrete k8s types? I'm not very familiar with this. If we can avoid it, it might be better though.

kyaml is useful primarily for mutating yaml. As you already discovered, the Marshal/Unmarshal functionality is just aliased.

Reply:
Switched to only accept Unstructured for now.
I think I may need support for typed objects later, but I don't yet. So i can avoid the runtime schema dependency, for now.